lib/Mongo/MongoCollection.php

```diff
@@ -772,5 +881,31 @@ private function ensureDocumentHasMongoId(&$document)
 
         return null;
     }
+
+    private function validateDocument(&$document)
```

Comment:
I think we can incorporate these checks into `ensureDocumentHasMongoId` since that will return false if the object given is not an object or array. We can just expand those checks to see if there are any empty documents there.